### PR TITLE
Bug 4343. fix "driver does not support quoting" when using the driver…

### PR DESCRIPTION
**Pull request by xpohoc69**
… pdo_odbc

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #4343

### CODE REVIEW

File: framework/db/CDbConnection.php [hunk 1]
```diff
@@ -575,11 +575,12 @@ public function quoteValue($str)
 		if(is_int($str) || is_float($str))
 			return $str;
 
-		$this->setActive(true);
-		if(($value=$this->_pdo->quote($str))!==false)
-			return $value;
-		else  // the driver doesn't support quote (e.g. oci)
-			return "'" . addcslashes(str_replace("'", "''", $str), "\000\n\r\\\032") . "'";
+        $this->setActive(true);
```
Comment (samdark) [inline framework/db/CDbConnection.php:578]: Formatting should use tabs. Need to fix that before merge.

**Review by samdark — APPROVED** [review]

File: framework/db/CDbConnection.php (hunk 1 above)
Comment (xpohoc69) [reply framework/db/CDbConnection.php:578]: @samdark fixed. used tabs now

**Comment by marcovtwout** [comment]
I plan to review this end of this week.

**Comment by xpohoc69** [comment]
@marcovtwout ok, thank you

**Comment by marcovtwout** [comment]
@samdark Two questions for you.

I suggest a slightly improved implementation where I refactored the duplicate code into a private function, and improved readaibility and yii1-formatting a bit: https://github.com/yiisoft/yii/commit/c5553696e554e24dfb90c8dbb4cdcc6ec88b7787

Also should we update yiilite with every change or only when creating a new release?

**Comment by samdark** [comment]
> I suggest a slightly improved implementation where I refactored the duplicate code into a private function, and improved readaibility and yii1-formatting a bit: c555369

OK.

> Also should we update yiilite with every change or only when creating a new release?

Usually it's release but updating right away won't hurt.

**Comment by marcovtwout** [comment]
Closing this in favor of https://github.com/yiisoft/yii/pull/4352

**Pull request closed without merging**